rust/otap-dataflow/crates/config/src/pipeline.rs

```diff
@@ -728,39 +729,75 @@ impl<'de> Deserialize<'de> for MetricsReaderConfig {
         }
 
         let reader_options_result = ReaderOptions::deserialize(deserializer);
-        if let Ok(options) = reader_options_result {
-            if let Some(config) = options.periodic {
-                Ok(MetricsReaderConfig::Periodic(config))
-            } else if options.pull.is_some() {
-                Ok(MetricsReaderConfig::Pull {})
-            } else {
-                Err(serde::de::Error::custom(
-                    "Expected either 'periodic' or 'pull' reader",
-                ))
+        match reader_options_result {
+            Ok(options) => {
+                if let Some(config) = options.periodic {
+                    Ok(MetricsReaderConfig::Periodic(config))
+                } else if options.pull.is_some() {
+                    Ok(MetricsReaderConfig::Pull {})
+                } else {
+                    Err(serde::de::Error::custom(
+                        "Expected either 'periodic' or 'pull' reader",
+                    ))
+                }
             }
-        } else {
-            Err(serde::de::Error::custom(
-                "Invalid configuration of a metrics reader",
-            ))
+            Err(err) => Err(serde::de::Error::custom(format!(
+                "Expected a map with either 'periodic' or 'pull' reader: {}",
+                err
+            ))),
         }
     }
 }
 
 /// Opentelemetry Metrics Perioidc Exporter configuration.
-#[derive(Debug, Clone, Serialize, JsonSchema)]
+#[derive(Debug, Clone, Serialize, JsonSchema, PartialEq)]
 pub enum MetricsPeriodicExporterConfig {
     /// Console exporter that writes metrics to the console.
-    Console(ConsoleExporterConfig),
+    Console,
     /// OTLP exporter that sends metrics using the OpenTelemetry Protocol.
-    Otlp,
+    Otlp(OtlpExporterConfig),
 }
 
-/// Opentelemetry Console Exporter configuration.
-#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
-pub struct ConsoleExporterConfig {
-    /// The temporality preference for the console exporter.
-    #[serde(rename = "temporality")]
-    pub temporality: Option<String>,
+/// Opentelemetry OTLP Exporter configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq)]
+pub struct OtlpExporterConfig {
+    /// The Otlp communication protocol to use when exporting data.
+    #[serde(default)]
+    pub protocol: OtlpProtocol,
+    /// The endpoint to which the Otlp exporter will send data.
+    pub endpoint: String,
+
+    /// The temporality of the metrics to be exported.
+    #[serde(default)]
+    pub temporality: Temporality,
```

Comment:
For future iteration - good to add header support - for auth headers (bearer token, API-KEYs etc). otel-otlp supports adding headers in builder, so should be feasible to extend.